src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs

```diff
@@ -96,8 +96,13 @@ public void StartForwarding(ScriptInvocationContext context, Uri httpUri)
             HttpContext httpContext = httpRequest.HttpContext;
             httpContext.Items[ScriptConstants.HttpProxyingEnabled] = bool.TrueString;
 
+            string invocationId = context.ExecutionContext.InvocationId.ToString();
+
             // add invocation id as correlation id
-            httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString());
+            if (!httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, invocationId))
```

Comment:
Just remove this whole `if` -- we're gonna overwrite whatever is there anyway, no need to check for it.